src/components/VmDetails/BaseCard.js

```diff
@@ -0,0 +1,130 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import {
+  Badge,
+  Button,
+  Card,
+  CardHeading,
+  CardTitle,
+  CardBody,
+  CardFooter,
+  Icon,
+  noop,
+  excludeKeys,
+} from 'patternfly-react'
+
+import style from './style.css'
+import CardEditButton from './CardEditButton'
+
+/**
+ * Base VM details card.  Support common layouts and view vs edit modes.
+ *
+ * Specifying __editMode__ allows the containing component to control the card's
+ * edit state.  Leave __editMode__ undefined to allow the card to control itself.
+ *
+ * When the user interacts with the card, the card will fire event handler:
+ *  - When in non-edit state, user clicks the __Edit__ icon -> onStartEdit() is called
+ *  - When in edit state, user clicks the __Cancel__ button -> onCancel() is called
+ *  - When in edit state, user clicks the __Save__ button -> onSave() is called
+ *
+ * If any of the event handlers return null, the card will not transition its edit state.
+ * This allows data validation or async operation completion.
+ */
+class BaseCard extends React.Component {
+  static propTypes = {
```

Comment:
Meh.  I like the static field syntax better.

However, since we use both class and function components, it does make sense to keep it with the old style static variable.  I'll move it down.

Reply:
+1 to moving it down, thanks